src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -93,10 +97,10 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             {
                 _fileWriter.AppendLine(formattedMessage);
 
-                // flush errors immediately
+                // we issue an immediate flush to ensure errors are logged as quickly as possible
                 if (logLevel == LogLevel.Error || exception != null)
                 {
-                    _fileWriter.Flush();
+                    _flushLogs();
```

Comment:
Previously every single error logged would result in a separate file write. These are now debounced to a max of 1 write every 300ms.

Reply:
Worth mentioning that with the debouncing, this is no longer going to be synchronous, which was another issue causing problems in the RPC logging path. 

Have you validated the behavior when an exception is thrown as a result, though? With your debounce change, that exception will now be thrown in an unobserved `Task`, and no longer caught by your exception handler below.

We need to make sure your debounced action is handling any exceptions previously handled in this block.